### PR TITLE
BOM-1015

### PR DESCRIPTION
**Pull request by awais786**
Adding tox for django 2, 2.1, 2.2 under python3.5

**Description:** Describe in a couple of sentence what this PR adds

**JIRA:** Link to JIRA ticket
https://openedx.atlassian.net/browse/BOM-1015

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Reviewers:**
- [ ] tag reviewer 
- [ ] tag reviewer 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: edx_when/models.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ class ContentDate(models.Model):
     """
 
     course_id = CourseKeyField(db_index=True, max_length=255)
-    policy = models.ForeignKey(DatePolicy)
+    policy = models.ForeignKey(DatePolicy, on_delete=models.CASCADE)
```
Comment (awais786) [inline edx_when/models.py]: django2 requirement

File: requirements/base.txt [hunk 2]
```diff
@@ -9,36 +9,39 @@ backports.os==0.1.1       # via fs
 certifi==2019.9.11        # via requests
 chardet==3.0.4            # via requests
 django-model-utils==3.2.0
-django-waffle==0.17.0     # via edx-django-utils, edx-drf-extensions
-django==1.11.24
+django-waffle==0.18.0     # via edx-django-utils, edx-drf-extensions
+django==1.11.26 ; python_version < "3"
```
Comment (jmbowman) [inline requirements/base.txt]: This isn't correct, so we may need to give up on the upper constraint on Django in `base.in` and just leave it as `Django>=1.11`; I don't think there's a good way to get `pip-compile` to generate what we want here with an environment marker involved (there's an [open issue](https://github.com/jazzband/pip-tools/issues/518) for this).  I've updated the [BOM-1009](https://openedx.atlassian.net/browse/BOM-1009) description accordingly.

File: tox.ini [hunk 3]
```diff
@@ -37,6 +37,8 @@ norecursedirs = .* docs requirements
 deps =
     django111: Django>=1.11,<2.0
     django20: Django>=2.0,<2.1
+    django21: Django>=2.1,<2.2
+    django22: Django>=2.2,<2.3
     -r{toxinidir}/requirements/test.txt
 commands =
     py.test {posargs}
```
Comment (jmbowman) [inline tox.ini]: To output any applicable deprecation warnings, this needs to be updated to `python -Wd -m pytest {posargs}`.

**Comment by awais786** [comment]
@jmbowman Please check again.

**Comment by awais786** [comment]
@jmbowman it shows some warnings.
https://travis-ci.org/edx/edx-when/jobs/608363291?utm_medium=notification&utm_source=github_status

**Comment by jmbowman** [comment]
That warning is originating in the edx-opaque-keys package, so it'll need to be updated also.  I just created [BOM-1017](https://openedx.atlassian.net/browse/BOM-1017) for that.

File: edx_when/models.py (hunk 1 above)
Comment (Ayub-Khan) [reply edx_when/models.py]: but are we sure we want to delete ?

**Comment by awais786** [comment]
This PR depends on this PR for warnings fixes https://github.com/edx/opaque-keys/pull/111

**Comment by awais786** [comment]
@jmbowman I have waited for opaque-key PR to resolve the warnings but I think its fine to merge this now and update opaque-key version in next PR after resolving the warnings. 
What do you think?

**Comment by awais786** [comment]
> @jmbowman I have waited for opaque-key PR to resolve the warnings but I think its fine to merge this now and update opaque-key version in next PR after resolving the warnings.
> What do you think?

@jmbowman any update?

**Comment by jmbowman** [comment]
Sure, that works for me.

**Pull request merged**